src/Core/tests/DeviceTests/TestBase.cs

```diff
@@ -18,5 +18,14 @@ public partial class TestBase
 
 		public Task<T> InvokeOnMainThreadAsync<T>(Func<Task<T>> func) =>
 			TestDispatcher.Current.DispatchAsync(func);
+
+		protected async Task WaitForGC()
+		{
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+			await Task.Delay(10);
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+		}
```

Comment:
One thing to mention is the Mono runtime team had us use this class:

https://github.com/xamarin/java.interop/blob/bc5bcf4f0ef07aab898f2643d2a25f66512d98ed/tests/Java.Interop-Tests/Java.Interop/FinalizerHelpers.cs

It fills the stack 128 methods deep, to make tests able to wait on the GC more reliably.

But I think you could probably wait and see if these tests fail randomly down the road.